addons/mail/static/src/core/common/activity_model.js

```diff
@@ -0,0 +1,77 @@
+import { Record } from "@mail/core/common/record";
+import { assignDefined } from "@mail/utils/common/misc";
+
+/**
+ * @typedef Data
+ * @property {string} activity_category
+ * @property {[number, string]} activity_type_id
+ * @property {string|false} activity_decoration
+ * @property {boolean} can_write
+ * @property {'suggest'|'trigger'} chaining_type
+ * @property {string} create_date
+ * @property {[number, string]} create_uid
+ * @property {string} date_deadline
+ * @property {string} date_done
+ * @property {string} display_name
+ * @property {boolean} has_recommended_activities
+ * @property {string} icon
+ * @property {number} id
+ * @property {Object[]} mail_template_ids
+ * @property {string} note
+ * @property {number|false} previous_activity_type_id
+ * @property {number|false} recommended_activity_type_id
+ * @property {string} res_model
+ * @property {[number, string]} res_model_id
+ * @property {number} res_id
+ * @property {string} res_name
+ * @property {number|false} request_partner_id
+ * @property {'overdue'|'planned'|'today'} state
+ * @property {string} summary
+ * @property {[number, string]} user_id
+ * @property {string} write_date
+ * @property {[number, string]} write_uid
+ */
```

Comment:
This typedef is not necessary/obsolete.

What we want instead is to define all the fields we return on the model itself.